### PR TITLE
python-common: Add more type annotations

### PR DESCRIPTION
**Pull request by sebastian-philipp**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/python-common/ceph/deployment/service_spec.py [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+from typing import Optional, Dict, Any, List, Union, Callable, Iterable, Type, TypeVar, cast, \
+    NamedTuple
 
 import yaml
 
 from ceph.deployment.hostspec import HostSpec
 from ceph.deployment.utils import unwrap_ipv6
 
+ServiceSpecT = TypeVar('ServiceSpecT', bound='ServiceSpec')
+FuncT = TypeVar('FuncT', bound=Callable)
```
Comment (tchaikov) [inline src/python-common/ceph/deployment/service_spec.py:15]:
```
ceph/deployment/service_spec.py:17:1: E302 expected 2 blank lines, found 1
```

**Review by tchaikov — APPROVED** [review]
might want to fix the lint warning.

File: src/python-common/ceph/deployment/service_spec.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/python-common/ceph/deployment/service_spec.py:15]: done

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-02-02_15:01:12-rados-wip-kefu-testing-2021-02-02-1939-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-02-03_08:10:00-rados-wip-kefu-testing-2021-02-03-1357-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-02-03_14:17:43-rados-wip-kefu-testing-2021-02-03-2051-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
do you plan to backport this to pacific?

**Pull request merged**